docs/docs/20-usage/40-secrets.md

```diff
@@ -48,6 +48,7 @@ This way, the secret key and environment variable name can differ.
 ### Using secrets in plugins-steps via "settings:"
 
 The `from_secret` syntax also work for settings in any hierarchy.
+Note that you can not use `secrets` and also not use it with `environment`, as this could alter the execution of plugins and so it is forbidden by the linter now.
```

Comment:
Thanks for taking a stab. However, I think we should expand on this and add more content. This is a very important point for users to understand the "why" behind this.

Maybe we can even put it into a highlight box?

Reply:
You must have a concrete idea in mind, could you push that onto by branch?

Reply:
Not really, I just think that these two assets are essential for many and removing them/explaining why one cannot/should not set secrets in or environment in plugins is essential - especially if the same are still allowed in normal steps.

You are better at explaining the reasoning in detail behind all of this. Happy to rephrase wording afterwards.

- Why (and how) could it alter plugin execution (Yes, I know there was a sec issue etc. but (many) users don't know)
- Add that secrets are essentially == environment and therefore both must be removed
- Explain why secrets defined as plugin settings are not harmful (this is actually still an open Q to me)